### DISCUSSION TITLE
AZURE AD ADAL not working after Quasar Upgrade

### DISCUSSION THREAD

**Discussion by arunima-nair** [quasarframework/quasar#13492] [Quasar App Extensions] [answered]
Hello team,
I upgraded quasar to version 
![image](https://user-images.githubusercontent.com/37262910/169745034-e91fe60b-7284-49e9-89f9-701b6b71fe2e.png)
We are using AZURE AD ADAL library for authentication. After upgradation (vue to vue 3) I am facing issues with this library as follows:
![image](https://user-images.githubusercontent.com/37262910/169745355-f2ead500-db4e-47bc-9dcc-010e03959ad8.png)
Please help me to resolve this . I am blocking with this error

**Comment by metalsadman** [comment, marked as answer]
we need to see the code where that is called.

**Reply by arunima-nair** [reply]
![](https://user-images.githubusercontent.com/37262910/169746598-957dcf07-3c7c-4665-82b0-715556e01ba7.png)
Above line showing error
![image](https://user-images.githubusercontent.com/37262910/169748172-58515068-e9ff-46d6-bc64-cf8279a722e6.png)

**Reply by metalsadman** [reply]
seem like you are using a vue 2 implementation, quasar v2 uses Vue 3, please see https://v3-migration.vuejs.org/breaking-changes/global-api.html#vue-prototype-replaced-by-config-globalproperties

I suggest reading everything in that docs as well, see if you need to migrate more parts of your code to Vue 3.

**Reply by arunima-nair** [reply]
I think ADAL library not yet migrated to VUE3

**Reply by arunima-nair** [reply]
![image](https://user-images.githubusercontent.com/37262910/169784033-73906c85-2191-405d-ab40-30f395cf96d0.png)
This is our code
